apps/btc_family/btc_txn_helpers.c

```diff
@@ -168,11 +168,12 @@ STATIC uint32_t get_transaction_weight(const btc_txn_context_t *txn_ctx) {
     weight += 32;    // previous transaction hash
     weight += 4;     // previous output index
     weight += 1;     // script length size
-    weight += txn_ctx->inputs[input_index].script_pub_key.size;
-    weight += 4;    // sequence
-                    // Check if current input is segwit or not
+    weight += 4;     // sequence
+                     // Check if current input is segwit or not
     if (0 == txn_ctx->inputs[input_index].script_pub_key.bytes[0]) {
       segwit_count++;
+    } else {
+      weight += txn_ctx->inputs[input_index].script_pub_key.size;
```

Comment:
let's remove this entirely. using `script_pub_key` is purely wrong. the `script_sig` is same for P2PKH and P2WPKH; and that is almost of fixed size (~106 bytes). let's use the same value of `script_sig` for both.

so, in summary, this function will give precise results for `P2PKH` and `P2WPKH` when we use a fixed size for `script_sig` (106 bytes). For P2PK and P2MS we will not have to bother as these are obsolete. For `P2SH` & `P2WSH`, the function will give wrong results so we will re-evaluate this function when the support for those types is supported.

Reply:
Refer: https://github.com/trezor/trezor-firmware/blob/f5983e7843f381423f30b8bc2ffc46e496775e5a/core/src/apps/bitcoin/sign_tx/tx_weight.py#L95

Refer: https://github.com/trezor/trezor-firmware/blob/f5983e7843f381423f30b8bc2ffc46e496775e5a/common/protob/messages-bitcoin.proto#L357

Reply:
added a script_sig_size variable for legacy transactions, also replaced them with p2pkh transactions, they were p2pk earlier